src/nni_manager/core/nniDataStore.ts

```diff
@@ -161,9 +165,8 @@ class NNIDataStore implements DataStore {
 
     private async getFinalMetricData(trialJobId: string): Promise<any> {
         const metrics: MetricDataRecord[] = await this.getMetricData(trialJobId, 'FINAL');
-        assert(metrics.length <= 1);
-        if (metrics.length === 1) {
-            return metrics[0];
+        if (metrics.length > 0) {
+            return metrics[metrics.length - 1];
```

Comment:
Should we also consider to log error if metrics.length > 1 for single phase exp?

Reply:
Ok, added back the assertion for single phase exp.